### DISCUSSION TITLE
erroneous extra indentation ?

### DISCUSSION THREAD

**Discussion by gst** [devpi/devpi#899] [General]
https://github.com/devpi/devpi/blob/4d3a4a35bf91d92c2a844c40fbca1d2938b445f7/server/devpi_server/model.py#L1514

should not it  be one indentation level down/less ?

```python
for link in self.get_links(rel=rel, basename=basename):
    # same as now

self.remove_links(rel=rel, basename=basename)
```

?

**Comment by fschulze** [comment]
Could be, do you see unexpected behavior that seems to be caused by this?

**Reply by gst** [reply]
only noticed by reading the code.

it looked overkill to remove the same links multiple time.

**Reply by gst** [reply]
But/and by the way : 

I don't understand the necessity/requirement of that actually. why links are being removed in **create_linked_entry** in the first time ?

I had to comment it in my current attempt to implement the snapshot feature I described in my issue..

not sure what's going on..

EDIT: ok I found out why I was having weird behavior (the wheel files of my source index snapshot were being removed because I re-use the verdata from the source index packages when I snapshot them into the target index...

I'm preparing an MR to show you what I got. in it I remove this line.

**Comment by fschulze** [comment]
The function seems to never apply to more than one link, so it doesn't matter much, but I changed it anyway. Thanks!
